src/main/java/pl/nyczjablonska/recursion/BinarySearch.java

```diff
@@ -8,13 +8,13 @@ public class BinarySearch {
     //    3. Zaimplementuj wyszukiwanie binarne na
 //    posortowanej tablicy intów o długości 100.
         public static int search(int[] array, int firstIndex, int lastIndex, int numberToFound) {
-        if (firstIndex < lastIndex) {
-            int middle = (firstIndex + lastIndex) / 2;
+        if (firstIndex <= lastIndex) {
+            int middle = firstIndex + (lastIndex - firstIndex) / 2;
```

Comment:
can you provide more insight on this equation: `firstIndex + (lastIndex - firstIndex) / 2`?
In case when `length = 8`, `firstIndex = 4` and `lastIndex = 7` then the equation will read `4 + ( 7 - 4 ) / 2 = 4 + (3/2) = 4 + 1 = 5`.. I wonder, if perhaps an easier equation, like `( lastIndex + firstIndex ) / 2 ` which would yield the same result ` ( 7 + 4 ) / 2 = 11 / 2 = 5` wouldn't be easier to understand...?